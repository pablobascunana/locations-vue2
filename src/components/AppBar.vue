<template>
  <v-app-bar id="appBar" app color="terciary" dark height="75">
    <h4>{{ $t('appBar.welcome') }} {{ userName }}</h4>
    <v-spacer></v-spacer>
    <v-icon id="logout" class="mr-3" @click="logout()">mdi-exit-to-app</v-icon>
  </v-app-bar>
</template>

<script>
import { mapState } from 'vuex';
export default {
  name: 'appBar',
  computed: {
    ...mapState('application', ['logoutDialogVisible']),
    ...mapState('user', ['userName'])
  },
  methods: {
    logout() {
      this.$store.commit('application/isLogoutDialogVisible', true);
    }
  }
}
</script>
