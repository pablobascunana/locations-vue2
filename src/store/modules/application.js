function initialApplicationState() {
  return {
    isLogin: true,
    accessToken: undefined,
    refreshToken: undefined,
    addLocationDialogVisible: false,
    removeLocationDialogVisible: false
  }
}

const application = {
  namespaced: true,
  state: initialApplicationState,
  mutations: {
    resetApplicationStore(state) {
      const initState = initialApplicationState();
      Object.keys(initState).forEach(key => {
        state[key] = initState[key];
      })
    },
    isLogin(state, visible) {
      state.isLogin = visible;
    },
    setAccessToken(state, token) {
      state.accessToken = `Bearer ${token}`;
    },
    setRefreshToken(state, token) {
      state.refreshToken = token;
    },
    isAddLocationDialogVisible(state, visible) {
      state.addLocationDialogVisible = visible;
    },
    isRemoveLocationDialogVisible(state, visible) {
      state.removeLocationDialogVisible = visible
    }
  }
};

export default application;
  