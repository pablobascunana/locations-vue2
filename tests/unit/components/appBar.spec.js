import { createLocalVue, shallowMount } from '@vue/test-utils'

import Vuetify from 'vuetify'
import Vuex from 'vuex'

import application from '@/store/modules/application.js'
import user from '@/store/modules/user.js'

import AppBar from '@/components/AppBar.vue'


const localVue = createLocalVue();
localVue.use(Vuex);
let vuetify = new Vuetify();


describe('AppBar', () => {
    let wrapper;
    let store;
    let state = {};
    let actions = {};
    let modules = {
        application,
        user
    };
    beforeEach(() => {
        store = new Vuex.Store({ state, actions, modules });

        wrapper = shallowMount(AppBar, {
            localVue,
            vuetify,
            store,
            mocks: {
                $t: () => {'some specific text in any language'}
            }
        });
    });

    it("Should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()    
    });

    it('Renders a vue instance', () => {
        expect(wrapper.isVueInstance()).toBe(true);
    });

    it('Call to logout function', () => {
        const logoutFn = jest.spyOn(wrapper.vm, 'logout');
        wrapper.vm.logout();
        expect(logoutFn).toBeCalled();
        expect(wrapper.vm.$store.state.application.logoutDialogVisible).toBe(true);
    });
});