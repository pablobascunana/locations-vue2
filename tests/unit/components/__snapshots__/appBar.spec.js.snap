// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Should match snapshot 1`] = `
<v-app-bar id="appBar" app="" color="terciary" dark="" height="75">
  <h4> function String() { [native code] }</h4>
  <v-spacer></v-spacer>
  <v-icon id="logout" class="mr-3">mdi-exit-to-app</v-icon>
</v-app-bar>
`;

exports[`AppBar should match snapshot 1`] = `
<v-app-bar id="appBar" app="" color="terciary" dark="" height="75">
  <h4> function String() { [native code] }</h4>
  <v-spacer></v-spacer>
  <v-icon id="logout" class="mr-3">mdi-exit-to-app</v-icon>
</v-app-bar>
`;
